fetchChallenges.js

```diff
@@ -1,64 +1,65 @@
-const res = await fetch(
-  "https://lernia-sjj-assignments.vercel.app/api/challenges"
-);
-const data = await res.json();
-export const challenges = data.challenges;
-if (window.location.pathname.includes("ourChallenges")) {
-  const container = document.querySelector(".ourChallenges");
-  renderChallenges(container);
+const data = await fetchData(`https://lernia-sjj-assignments.vercel.app/api/challenges`).catch(error => console.log(error.message))
+
+async function fetchData(url) {
+    const response = await fetch(url)
+    if(response.ok) {
+        return await response.json();
+    } else {
+        throw new Error(`Something went wrong with the request. Error code: ${response.status}`)
+    }
 }
 
-export function renderChallenges(container, threeHighest) {
-  let challengesToRender;
-  if (threeHighest) {
-    challengesToRender = challenges
-      .sort((a, b) => b.rating - a.rating)
-      .slice(0, 3);
-  } else {
-    challengesToRender = challenges;
-  }
-  const challengeTemplate = document.getElementById("card-template");
-  challengesToRender.forEach((challenge) => {
-    const template = challengeTemplate.content.cloneNode(true);
-    const card = template.querySelector("article");
-    const image = template.querySelector(".sidescroll__img");
-    const icon = template.querySelector(".sidescroll__icon");
-    const h2 = template.querySelector("h2");
-    const rating = template.querySelector(".sidescroll__rating");
-    const participans = template.querySelector(".sidescroll__participants");
-    const description = template.querySelector(".sidescroll__text");
-    const button = template.querySelector(".sidescroll__btn");
-    button.textContent = challenge.type === "online" ? "Take challenge online" : "Book this room";
-    icon.classList.add(challenge.type === "online" ? "fa-laptop" : "fa-house");
-    card.id = challenge.id;
-    image.src = challenge.image; // `${challenge.image}?image=${Math.floor(Math.random() * 16)}` to make image random
-    h2.textContent = challenge.title;
-    participans.textContent =
-      challenge.minParticipants === challenge.maxParticipants
-        ? `${challenge.minParticipants} participants`
-        : `${challenge.minParticipants}-${challenge.maxParticipants} participants`;
-    description.textContent = challenge.description;
-    createStars(rating.firstElementChild, challenge.rating);
-    container.appendChild(template);
-  });
+export const challenges = data?.challenges;
+if (window.location.pathname.includes('ourChallenges')) {
+    const container = document.querySelector('.ourChallenges')
+    renderChallenges(container)
+} else {
+    const container = document.querySelector('.sidescroll')
+    renderChallenges(container, true)
```

Comment:
Boolean parameters like `threeHighest` are often frowned upon, because it's difficult to understand what they do. When I read this line, it's very difficult for me to know what the difference is between passing `true` or not. Reading the function makes it easier, but still not obvious.